feature/settings/src/main/java/com/google/jetpackcamera/settings/ui/SettingsComponents.kt

```diff
@@ -240,6 +243,92 @@ fun CaptureModeSetting(currentCaptureMode: CaptureMode, setCaptureMode: (Capture
+                    fontStyle = FontStyle.Italic,
+                    color = MaterialTheme.colorScheme.onPrimaryContainer
+                )
+                Spacer(modifier = Modifier.height(10.dp))
+                SingleChoiceSelector(
+                    text = stringResource(id = R.string.stabilization_selector_on),
+                    selected = currentPreviewStabilization == Stabilization.ON,
+                    onClick = { setPreviewStabilization(Stabilization.ON) }
+                )
+                SingleChoiceSelector(
+                    text = stringResource(id = R.string.stabilization_selector_off),
+                    selected = currentPreviewStabilization == Stabilization.OFF,
+                    onClick = { setPreviewStabilization(Stabilization.OFF) }
+                )
+                SingleChoiceSelector(
+                    text = stringResource(id = R.string.stabilization_selector_undefined),
+                    selected = currentPreviewStabilization == Stabilization.UNDEFINED,
+                    onClick = { setPreviewStabilization(Stabilization.UNDEFINED) }
+                )
+            }
+        }
+    )
+}
+
+@Composable
+fun VideoStabilizeSetting(
+    currentPreviewStabilization: Stabilization,
+    setVideoStabilization: (Stabilization) -> Unit
+) {
+    BasicPopupSetting(
+        title = stringResource(R.string.video_stabilization_title),
+        leadingIcon = null,
+        description = when (currentPreviewStabilization) {
+            Stabilization.UNDEFINED -> stringResource(
+                id = R.string.stabilization_description_undefined
+            )
+            Stabilization.OFF -> stringResource(id = R.string.stabilization_description_off)
+            Stabilization.ON -> stringResource(id = R.string.stabilization_description_on)
+        },
+        popupContents = {
+            Column(Modifier.selectableGroup()) {
+                Text(
+                    text = stringResource(id = R.string.stabilization_popup_disclaimer),
+                    fontStyle = FontStyle.Italic,
+                    color = MaterialTheme.colorScheme.onPrimaryContainer
+                )
+                Spacer(modifier = Modifier.height(10.dp))
+                SingleChoiceSelector(
+                    text = stringResource(id = R.string.stabilization_selector_on),
+                    selected = currentPreviewStabilization == Stabilization.ON,
+                    onClick = { setVideoStabilization(Stabilization.ON) }
+                )
+                SingleChoiceSelector(
+                    text = stringResource(id = R.string.stabilization_selector_off),
+                    selected = currentPreviewStabilization == Stabilization.OFF,
+                    onClick = { setVideoStabilization(Stabilization.OFF) }
+                )
+                SingleChoiceSelector(
+                    text = stringResource(id = R.string.stabilization_selector_undefined),
```

Comment:
While I can appreciate trying to mirror the APIs, we should consider that the APIs are meant for developers whereas UI is meant for app users. An app user probably has no idea what "`UNDEFINED`" means.

Looking at the table:
![image](https://github.com/google/jetpack-camera-app/assets/1455403/7d658a11-2e85-4c2f-9d24-23f38d508f7a)

Really, there are only 3 options that users should care about:
1. Preview AND the recorded video are stabilized (WYSIWYG) [rows 1, 3]
2. "High quality" recorded video stabilization (Preview _might_ be stabilized, depending on device) [row 7]
3. Stabilization off [all other rows]

I would only display these 3 options in the UI (we could call them `ON`, `HIGH QUALITY`, `OFF`), but add longer descriptions underneath each option to explain the difference between `ON` and `HIGH_QUALITY`.